Sources/SwiftInspectorVisitors/TypeSyntaxVisitor.swift

```diff
@@ -28,37 +28,39 @@ public final class TypeSyntaxVisitor: SyntaxVisitor {
 
   // MARK: Lifecycle
 
-  public init(typeName: String, onNodeVisit: @escaping (_ info: TypeProperties) -> Void) {
-    self.onNodeVisit = onNodeVisit
+  public init(typeName: String) {
     self.typeName = typeName
   }
 
   // MARK: Public
 
+  /// Information about each of the properties found on the type. `nil` if the type is not found.
+  public private(set) var propertiesData: Set<PropertyData>?
+
   public override func visit(_ node: ClassDeclSyntax) -> SyntaxVisitorContinueKind {
     if node.identifier.text == typeName {
-      processNode(node, withName: node.identifier.text, members: node.members.members)
```

Comment:
Since we are only trying to look up a specific type and we are only invoking `processNode(...)` when we find that type, let's simplify.

Reply:
I don't fully understand why `TypeSyntaxVisitor` is scoped to a single type name. I know it was doing that before, but... I don't fully get the use case. A `///` comment on `TypeSyntaxVisitor` might be helpful. I think it's finding the names of all declared types in a particular tree. Maybe what I'm looking for more than a `///` is a rename: `DeclaredTypeSyntaxVisitor`?

Open to thoughts here. None of these ideas are blocking since we're improving what was already here.

Reply:
Re: `DeclaredTypeSyntaxVisitor` it will find properties in an extension of a type too, [even if the extension is not on the type's declaration](https://github.com/fdiaz/SwiftInspector/blob/161075911041a9fbd24dc5f18dc042d10658be9d/Sources/SwiftInspectorVisitors/Tests/TypeSyntaxVisitorSpec.swift).

Reply:
I also want to highlight [this test](https://github.com/fdiaz/SwiftInspector/blob/161075911041a9fbd24dc5f18dc042d10658be9d/Sources/SwiftInspectorVisitors/Tests/TypeSyntaxVisitorSpec.swift#L133-L166). `TypeSyntaxVisitor` will find all properties of a non fully qualified type name in a file, even if those properties are on multiple distinct types.

Reply:
I think it would be really nice if we updated this visitor to accept a fully qualified type name. Until then, maybe we start with some documentation? 3ac70f3

Reply:
dig the doc comment